airbyte-cdk/python/airbyte_cdk/sources/streams/abstract_stream.py

```diff
@@ -0,0 +1,213 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from abc import ABC, abstractmethod
+from functools import lru_cache
+from typing import Any, Iterable, List, Mapping, Optional, Tuple, Union
+
+import requests
+from airbyte_cdk.models import AirbyteMessage, AirbyteStream, SyncMode
+from airbyte_cdk.models import Type as MessageType
+from airbyte_cdk.sources.source import Source
+from airbyte_cdk.sources.utils import casing
+from airbyte_cdk.sources.utils.schema_helpers import InternalConfig
+from airbyte_cdk.sources.utils.slice_logger import SliceLogger
+from airbyte_cdk.sources.utils.transform import TransformConfig, TypeTransformer
+from airbyte_cdk.sources.utils.types import JsonType, StreamData
+
+
+class AbstractStream(ABC):
+    @abstractmethod
+    def read(
```

Comment:
my feeling is that the magic wand input to this method should be `ConfiguredAirbyteStream`, `State`, and mabe a config. WDYT?

Reply:
I don't think either are required since both are known at initialization time. I suspect we might still want to pass the state to avoid the need to make massive changes to the abstract source, but I would prefer waiting until we support incremental reads to add the parameter